### PR TITLE
docs: Update OpenBSD build instructions for OpenBSD 6.3

### PR DESCRIPTION
**Pull request by practicalswift**
Update OpenBSD build instructions for OpenBSD 6.3.

### CODE REVIEW

File: doc/build-openbsd.md [hunk 1]
```diff
@@ -12,11 +12,10 @@ Preparation
 Run the following as root to install the base dependencies for building:
 
 ```bash
-pkg_add git gmake libevent libtool
-pkg_add autoconf # (select highest version, e.g. 2.69)
-pkg_add automake # (select highest version, e.g. 1.15)
-pkg_add python # (select highest version, e.g. 3.6)
-pkg_add boost
+pkg_add git gmake libevent libtool boost
```
Comment (fanquake) [inline doc/build-openbsd.md]: I don't think we need to specify minor/patch versions here. They are just more numbers that will quickly go "out of date"

Comment (practicalswift) [reply doc/build-openbsd.md]: Updated. Please re-review :-)

**Comment by laanwj** [comment]
Will test this soon.

**Comment by fanquake** [comment]
utACK 1680b8b, this is just making the documentation more verbose.

**Pull request merged**